staging/src/k8s.io/kubectl/pkg/describe/describe_test.go

```diff
@@ -2516,8 +2519,8 @@ func TestDescribeDeployment(t *testing.T) {
 			expects: []string{
 				"Replicas:           2 desired | 1 updated | 3 total | 2 available | 1 unavailable",
 				"Image:        mytest-image:v2.0",
-				"OldReplicaSets:  bar-001 (2/2 replicas created)",
-				"NewReplicaSet:   bar-002 (1/1 replicas created)",
+				"OldReplicaSets:    bar-001 (2/2 replicas created)",
```

Comment:
When updating the above, please make sure to also address this one.

Reply:
Hey @mimowo, the tests fail if I don't add the extra spaces here. I think it's due to the alignment of tabs with `Node-Selectors`. See:

```
--- FAIL: TestDescribeDeployment (0.00s)                                                                                                      
    --- FAIL: TestDescribeDeployment/deployment_during_the_process_of_rolling_out (0.00s) 
        describe_test.go:2839: expected to find "OldReplicaSets:  bar-001 (2/2 replicas created)" in:                
             Name:               bar                                                                                                          
            Namespace:          foo                                                                                                           
            CreationTimestamp:  Fri, 01 Jan 2021 00:00:00 +0000                                                                               
            Labels:             k8s-app=bar                                                                                                   
            Annotations:        <none>
            Selector:           k8s-app=bar
            Replicas:           2 desired | 1 updated | 3 total | 2 available | 1 unavailable
            StrategyType:                  
            MinReadySeconds:    0     
            Pod Template:                  
              Labels:  k8s-app=bar                                                                                                            
              Containers:       
               :                 
                Image:        mytest-image:v2.0
                Port:         <none>
                Host Port:    <none>
                Environment:  <none>
                Mounts:                        
                  /tmp/vol-bar from vol-bar (rw)
                  /tmp/vol-foo from vol-foo (rw)
              Volumes:              
               vol-foo:
                Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
                Medium:                         
                SizeLimit:  <unset>
               vol-bar:
                Type:          EmptyDir (a temporary directory that shares a pod's lifetime)
                Medium:        
                SizeLimit:     <unset>
              Node-Selectors:  <none>
              Tolerations:     <none>                                                                                                         
            OldReplicaSets:    bar-001 (2/2 replicas created)
            NewReplicaSet:     bar-002 (1/1 replicas created)
            Events:                  
              Type    Reason             Age                From                   Message
              ----    ------             ----               ----                   -------
              Normal  ScalingReplicaSet  12m (x3 over 20m)  deployment-controller  Scaled up replica set bar-002 to 1
              Normal  ScalingReplicaSet  10m                deployment-controller  Scaled up replica set bar-001 to 2
              Normal  ScalingReplicaSet  2m                 deployment-controller  Scaled up replica set bar-002 to 1
              Normal  ScalingReplicaSet  60s                deployment-controller  Scaled down replica set bar-002 to 1
```

Reply:
I see, indeed the values for `Node-Selectors` `Tolerations` and `OldReplicaSets` are aligned in the full output. 

Out of curiosity, I'm wondering where is the code / mechanism which determines the shift of `OldReplicaSets` to align it. Can you find it out?  So that we better understand the intent, because some values are out of alignment, for example `Volumes.vol-foo.Type`.

Reply:
This is the code for the empty dir volume: https://github.com/kubernetes/kubernetes/blob/d2b16b2/staging/src/k8s.io/kubectl/pkg/describe/describe.go#L1028-L1039
From reading the code, the value should be aligned, as it seems just a tab. This is the code for `OldReplicaSets`: https://github.com/kubernetes/kubernetes/blob/d2b16b2/staging/src/k8s.io/kubectl/pkg/describe/describe.go#L4372-L4375